fix: create errors orders and fix translations

Nortsova opened the pull request:
## Description
- Validation warnings should appear in the same order as the fields on the form (e.g., Title, From, Recipient, Amount, Decision Method) to ensure consistency and clarity.
- Each warning message should begin with a capital letter for consistency and readability.
- The pointer cursor should appear when hovering over elements. 

![cursor-pointer](https://github.com/user-attachments/assets/fb0278ca-7599-4c4f-af40-a4f80c56fffb)

## Testing
Step 1. Install voting reputation
Step 2. Install staged payments
Step 3. Open all action forms one by one and submit it to trigger validation
Step 4. Ensure that all form errors are consistent and shows in the same order as inputs

<img width="683" alt="image" src="https://github.com/user-attachments/assets/7d9f43ab-040d-458c-adc1-2fd0eb2f9d51" />

<img width="695" alt="image" src="https://github.com/user-attachments/assets/4f583fa9-eb1a-4af8-bf54-275502ace389" />

<img width="692" alt="image" src="https://github.com/user-attachments/assets/ad4b0e1d-b4c4-4096-b0dd-47c31def692c" />

<img width="677" alt="image" src="https://github.com/user-attachments/assets/d931f6aa-6b98-4c57-b910-03310bc9e3bd" />

<img width="677" alt="image" src="https://github.com/user-attachments/assets/3fca8715-b774-4369-871c-c42d13b90524" />

<img width="679" alt="image" src="https://github.com/user-attachments/assets/138b4da1-67de-4799-b82c-5fd3a54f8e1b" />

<img width="677" alt="image" src="https://github.com/user-attachments/assets/9720dd9c-19f8-481d-84f3-254a11ebc9ba" />

Step 5. Test that cursor pointer appears over the input labels:

![cursor-pointer](https://github.com/user-attachments/assets/fb0278ca-7599-4c4f-af40-a4f80c56fffb)

## Diffs

**New stuff** ✨

* **InputsOrderContext** to track in what order input was rendered

**Changes** 🏗

* Decision method field changed from defined() to required() to implement custom translations



Resolves #3737
Resolves #3743

mmioana reviewed (dismissed):
Really nice work @Nortsova 🥇 Thanks for fixing up the missing translations, it was something that was bugging me too for some time 😅 

From what I've tested all error messages are now showing up in the right order thanks to your solution of using the context for registering the inputs order 🌟 

Spam of screenshots on the way 
![Screenshot 2025-01-30 at 10 01 50](https://github.com/user-attachments/assets/a763e417-c252-4a71-9be1-5f2944b9011c)
![Screenshot 2025-01-30 at 10 05 10](https://github.com/user-attachments/assets/9e334313-b46a-4a40-bb4e-9c2d133e67a7)
![Screenshot 2025-01-30 at 10 06 01](https://github.com/user-attachments/assets/059ff930-6769-44bd-b600-2132b9c65375)
![Screenshot 2025-01-30 at 10 06 18](https://github.com/user-attachments/assets/681f75c9-2cb6-4a11-86f5-d968037e222f)
![Screenshot 2025-01-30 at 10 07 06](https://github.com/user-attachments/assets/77a5974b-1433-48a6-904f-374b46ccad57)
![Screenshot 2025-01-30 at 10 07 21](https://github.com/user-attachments/assets/0f205cb7-0bdc-402c-9a7c-ca83625fce1c)
![Screenshot 2025-01-30 at 10 07 28](https://github.com/user-attachments/assets/ccdfa435-2435-4ef5-8546-268a94371639)
![Screenshot 2025-01-30 at 10 07 35](https://github.com/user-attachments/assets/061c6cc5-6edf-4e09-814a-c566f30730a7)
![Screenshot 2025-01-30 at 10 07 54](https://github.com/user-attachments/assets/a1e75f50-b4af-4e50-9c5c-332fe4ae4220)
![Screenshot 2025-01-30 at 10 08 12](https://github.com/user-attachments/assets/a83b391a-70e1-4467-9c17-060ce66b00ab)
![Screenshot 2025-01-30 at 10 08 35](https://github.com/user-attachments/assets/f6c4c570-b879-4638-84a4-1d7788a18b06)
![Screenshot 2025-01-30 at 10 08 47](https://github.com/user-attachments/assets/cc1d0b55-0425-417e-ae9d-78e4e553142d)
![Screenshot 2025-01-30 at 10 08 55](https://github.com/user-attachments/assets/c609dcff-0824-43c9-97cf-87e81fc0c4c6)
![Screenshot 2025-01-30 at 10 09 09](https://github.com/user-attachments/assets/39a73b67-11f7-427b-a526-4c7018a2ccb8)
![Screenshot 2025-01-30 at 10 09 17](https://github.com/user-attachments/assets/61ee10e4-13e4-4c9d-aa3d-b5958ae93855)
![Screenshot 2025-01-30 at 10 09 59](https://github.com/user-attachments/assets/561d4563-c242-4218-9a16-bb4dbf2e6009)
![Screenshot 2025-01-30 at 10 11 06](https://github.com/user-attachments/assets/c077ae13-0515-4e1c-88f8-0911a99848a4)

And the input field descriptions also had a tooltip

https://github.com/user-attachments/assets/769b08b3-8067-4960-8311-8d95050de02e

Left a few refactoring comments, but nothing to block the merging of this PR, amazing work! 💯

Nortsova commented on src/components/v5/common/ActionSidebar/partials/ActionSidebarContent/hooks.ts:
```diff
@@ -15,7 +17,19 @@ export const useGetFormActionErrors = () => {
+    if (bIndex === -1) return -1;
+
+    return aIndex - bIndex;
+  });
```
good point! thank you :)

iamsamgibbs reviewed (dismissed):
Nice job! I like this solution! Approving as it does resolve the original issue, I've got a few observations I'll leave but they're not necessarily directly related so feel free to decide if you think they are worth tackling here.

The biggest simple change that stood out to me is that all the fields have popovers now except for "Action type" and  the "Token" field on the unlock token action. It would be nice to add those just for the sake of consistency.

I was able to get some wrong invalid errors showing. This only happens if you have the form open on "Staged payments" then switch to "Create agreement" and submit the form. If you directly open "Create agreement" via the sidebar, or select it after another action it isn't an issue.

![Screenshot 2025-01-31 at 09 33 05](https://github.com/user-attachments/assets/95cc3546-4a2a-468e-afe0-ef703e4402e2)

Also on the create agreement form, "description" is a required field but the placeholder does not go red when submitting.

And lastly there is a little inconsistency with the "Member" field on the "Manage reputation" form. The field is disabled so when the form is submitted, the label stays greyed out whilst the select changes to red. I think the intended behaviour here is for the select to also stayed greyed out whilst it is disabled.

![Screenshot 2025-01-31 at 09 36 04](https://github.com/user-attachments/assets/09b04ede-07da-4655-92c4-c75a7c2afe8c)

Nice job resolving the error order issue though, not a straightforward issue to solve!

rdig reviewed (changes requested):
Very nice work cleaning this up. I've tested all that I could, with most of them working. The only two that I had problems with are _Split Payments_ and _Manage Verified Members_ _(details below)_

**Working:**

1. Simple Payment ![Screenshot from 2025-01-31 14-28-12](https://github.com/user-attachments/assets/21d87b20-6020-4532-a3ee-24ec4b116545)
2. Advanced Payments ![image](https://github.com/user-attachments/assets/ca12e1b2-511f-4733-81fb-68469202b88b)
3. Staged Payments ![image](https://github.com/user-attachments/assets/adfd749a-1a7a-4e56-a167-3f4b5dd5dca3)
4. Transfer funds ![image](https://github.com/user-attachments/assets/861e4391-5ad1-4679-90ce-97d7760ba5b3)
5. Mint Tokens ![image](https://github.com/user-attachments/assets/8a88413f-b13b-4d55-8a42-1a2f166367b0)
6. Unlock Tokens ![image](https://github.com/user-attachments/assets/775aac6e-015b-4a08-b8fe-32a9a495f466)
7. Manage Tokens ![image](https://github.com/user-attachments/assets/89081970-93ae-429f-bc0b-e057873e45c4)
8. Create new domain ![image](https://github.com/user-attachments/assets/e3df9a7a-40b5-484a-9e71-67a67d94cf14)
9. Edit an existing domain ![image](https://github.com/user-attachments/assets/f8967d48-6b10-4a27-b21f-0fbed6058336)
10. Manage reputation ![image](https://github.com/user-attachments/assets/d2653861-7f47-4651-9b16-6e51ec888ba8)
11. Manage permissions ![image](https://github.com/user-attachments/assets/a0011eca-e7b9-4f89-b3b0-c414fbfb207a)
12. Arbitrary TX ![image](https://github.com/user-attachments/assets/cc8a679c-3b72-46ba-b4d3-d000a0b7cd63)
13. Edit Colony ![image](https://github.com/user-attachments/assets/70f731fd-58e7-4389-be0f-a5ba39a4e00b)
15. Create agreement ![image](https://github.com/user-attachments/assets/aee48ab9-8033-4f81-8319-d746fd95d564)

**Not working**

1. Split payment has recepient before percent, and doesn't have the individual amount _(although I'm not sure the last one is required)_ ![image](https://github.com/user-attachments/assets/0254ed98-010c-4a3f-966d-c4b74be47b0c)
2. Manage verified memebers has one extra entry _(I think it's for the same thing -- selecting the member)_ ![image](https://github.com/user-attachments/assets/911db8f2-c8b4-4c99-9548-0071a4cc4584)

Nortsova commented:
Thanks, @rdig , you were right; split payments were working incorrectly; I added `InputsOrderCellWrapper` to solve it.

<img width="701" alt="image" src="https://github.com/user-attachments/assets/2090d02b-1a68-40f0-bb60-c6b9873201e0" />


Regarding Manage verified members it is a bit complecated, it actually was a correct behaviour, because after user select method add or remove members table will appear, so last error was related to this hidden table. 
But it is not understandable for user, so I hided this error untill Method selected 🙌 

<img width="697" alt="image" src="https://github.com/user-attachments/assets/3aff9660-b0e9-465e-952c-079e3447cbcf" />
<img width="699" alt="image" src="https://github.com/user-attachments/assets/9555ddf0-95a2-4eef-a1b5-8e1cb1f1f067" />

rdig reviewed (dismissed):
All good now. Nicely done!

iamsamgibbs reviewed (dismissed):
Nice fixes!

Recipient and percentage are in the correct order:
![Screenshot 2025-02-04 at 10 55 12](https://github.com/user-attachments/assets/c9bd91e6-e816-4258-8cbd-cdfe4867fc57)

No "Select a member" error when "Add"/"Remove" has not yet been selected:
![Screenshot 2025-02-04 at 10 55 34](https://github.com/user-attachments/assets/17084210-0aa5-49c6-b764-9d2e5698daa4)

And then does show once selected and resubmitted:
![Screenshot 2025-02-04 at 10 55 48](https://github.com/user-attachments/assets/67ff9192-4df2-47f8-9d2b-64f47e897c84)

mmioana reviewed (changes requested):
Hey @Nortsova really amazing efforts on this issue 🙌 

I've retested the scenarios that were previously mentioned and it seems they behave properly 💯 
![Screenshot 2025-02-04 at 18 41 00](https://github.com/user-attachments/assets/de9129c7-53be-4b78-9ddf-f604c5fd7cb5)
![Screenshot 2025-02-04 at 18 41 22](https://github.com/user-attachments/assets/cfef8405-7156-4ed1-aa6a-b23ff2722990)
![Screenshot 2025-02-04 at 18 42 17](https://github.com/user-attachments/assets/038ae646-08f8-4ee8-99b9-e60caec14d50)

However @Nortsova I did encounter an issue when switching from `Staged payments` to `Manage verified members`, when the `createdIn` field appears as required and you can no longer submit the form 😢 

https://github.com/user-attachments/assets/178f3d05-e0f4-49bb-8bd9-9b598db7f12c

https://github.com/user-attachments/assets/c7d6b77a-1966-41db-90e3-b0f6d9a9aae6

Nortsova commented:
Wow, @mmioana , really nice catch! There was something wrong with the Staged payments form, and it caused this bug for all forms with the default `createdIn` field. Finally, I found a fix, but it still looks strange to me 😄 

![staged-payments-bug](https://github.com/user-attachments/assets/ea326c8a-ee7a-4fed-aca2-a59ab48e5aba)

rdig reviewed (approved):
Re-approving

mmioana reviewed (approved):
@Nortsova thanks for fixing the issue with `createdIn`. You did an amazing job on this PR! 🌟 


https://github.com/user-attachments/assets/b5c33488-d6e4-42a1-99e0-d64bb1d1d2b6
